netbox/extras/models/scripts.py

```diff
@@ -182,3 +183,10 @@ def save(self, *args, **kwargs):
 @receiver(post_save, sender=ScriptModule)
 def script_module_post_save_handler(instance, created, **kwargs):
     instance.sync_classes()
+
+
+@receiver(post_sync, sender=DataSource)
+def script_data_source_sync_handler(instance, **kwargs):
```

Comment:
We do not want to automatically sync files for which auto-sync has not been explicitly enabled. (That is the purpose of AutoSyncRecord.)